### ISSUE TITLE
New Rule: In ON clause of JOIN statement specify source table first and join table second

### ISSUE THREAD

**Issue by jpy-git** [sqlfluff/sqlfluff#2395]
From GitLab SQL style guide: https://about.gitlab.com/handbook/business-technology/data-team/platform/sql-style-guide/#joins

`Specify the order of a join with the FROM table first and JOIN table second:`
```sql
  -- Good
  FROM source
  LEFT JOIN other_source
    ON source.id = other_source.id
  WHERE ...

  -- Bad
  FROM source
  LEFT JOIN other_source
    ON other_source.id = source.id
  WHERE ...
```

More generally put the SOURCE table first and JOIN table second (since the source isn't always necessarily with a FROM clause):
```sql
  -- Good
  FROM source
  LEFT JOIN other_source
    ON source.id = other_source.id
  LEFT JOIN different_source
    ON other_source.id = different_source.id
  WHERE ...

  -- Bad
  FROM source
  LEFT JOIN other_source
    ON other_source.id = source.id
  LEFT JOIN different_source
    ON different_source.id = other_source.id
  WHERE ...
```

The motivation is that we already know the JOIN table from the JOIN clause so we want to be able to quickly scan for the SOURCE table.

**Comment by fmms** [comment]
though i personally prefer the other way arround, it would be awesome to have such a rule. Consistent ordering in ONs is really appreciated...

**Comment by thibault-he-cko** [comment]
I would also love to have such a rule, it's definitely one of the rules from the [dbt Style Guide](https://github.com/dbt-labs/corp/blob/main/dbt_style_guide.md#sql-style-guide) that I hate seeing breached...
